langserver/hover.go

```diff
@@ -208,23 +229,44 @@ func (s *server) getMetricDocs(ctx context.Context, metric string) (string, erro
 
 	api := v1.NewAPI(s.prometheus)
 
-	metadata, err := api.TargetsMetadata(ctx, "", metric, "1")
-	if err != nil {
-		return ret.String(), err
-	} else if len(metadata) == 0 {
-		return ret.String(), nil
-	}
+	if !compatibleVersion(s.PrometheusVersion, "2.15") {
+		metadata, err := api.TargetsMetadata(ctx, "", metric, "1")
+		if err != nil {
+			return ret.String(), err
+		} else if len(metadata) == 0 {
+			return ret.String(), nil
+		}
 
-	if metadata[0].Help != "" {
-		fmt.Fprintf(&ret, "__Metric Help:__ %s\n\n", metadata[0].Help)
-	}
+		if metadata[0].Help != "" {
```

Comment:
You could avoid the duplicate `Fprintf` statements by introducing local variables for `help`, `type` and `unit`.
(nit)